lib/std/setutils.nim

```diff
@@ -61,3 +61,17 @@ proc complement*[T](s: set[T]): set[T] {.inline.} =
     assert complement({range[0..10](0), 1, 2, 3}) == {range[0..10](4), 5, 6, 7, 8, 9, 10}
     assert complement({'0'..'9'}) == {0.char..255.char} - {'0'..'9'}
   fullSet(T) - s
+
+proc `[]=`*[T](t: var set[T], key: T, val: bool) {.inline.} =
+  ## Syntax sugar for `if val: t.incl key else: t.excl key`
```

Comment:
```suggestion
  ## Syntax sugar for `if val: t.incl(key) else: t.excl(key)`
```
I think this makes it more readable.

Reply:
controversial, i think (or would need to be settled in nep1)